test/acceptance/user.controller.acceptance.ts

```diff
@@ -107,6 +110,23 @@ describe('UserController', () => {
     await client.get(`/users/${newUser.id}`).expect(200, newUser.toJSON());
   });
 
+  describe('user product recommendation (service) api', () => {
+    // tslint:disable-next-line:no-any
+    let recommendationService: Server;
+
+    before(() => {
+      recommendationService = createRecommendationServer();
+    });
+
+    after(() => {
+      recommendationService.close();
+    });
+
+    it('returns product recommendations for a user', async () => {
+      await client.get(`/users/userid/recommend`).expect(200, recommendations);
```

Comment:
I'm probably missing something, but shouldn't `userid` be a number here?

Reply:
Mongo generated UUID for User model.